Framework/Algorithms/test/ConvertTableToMatrixWorkspaceTest.h

```diff
@@ -139,6 +139,88 @@ class ConvertTableToMatrixWorkspaceTest : public CxxTest::TestSuite {
     TS_ASSERT_THROWS_NOTHING(m_converter->setPropertyValue("ColumnX", "A"));
     TS_ASSERT_THROWS_NOTHING(m_converter->setPropertyValue("ColumnY", "B"));
   }
+  // test if it can convert a string to a double, if the string is numeric
+  void testStringToDouble() {
+
+    ITableWorkspace_sptr tws = WorkspaceFactory::Instance().createTable();
+    tws->addColumn("str", "A");
+    tws->addColumn("double", "B");
+    tws->addColumn("double", "C");
+
+    size_t n = 10;
+    for (size_t i = 0; i < n; ++i) {
+      TableRow row = tws->appendRow();
+      std::string x = "1";
+      double y = double(i) * 1.1;
+      double e = sqrt(y);
+      row << x << y << e;
+    }
+
+    TS_ASSERT_THROWS_NOTHING(m_converter->setProperty("InputWorkspace", tws));
+    TS_ASSERT_THROWS_NOTHING(
+        m_converter->setPropertyValue("OutputWorkspace", "out"));
+    TS_ASSERT_THROWS_NOTHING(m_converter->setPropertyValue("ColumnX", "A"));
+    TS_ASSERT_THROWS_NOTHING(m_converter->setPropertyValue("ColumnY", "B"));
+    TS_ASSERT_THROWS_NOTHING(m_converter->setPropertyValue("ColumnE", "C"));
+
+    TS_ASSERT(m_converter->execute());
+
+    MatrixWorkspace_sptr mws = boost::dynamic_pointer_cast<MatrixWorkspace>(
+        API::AnalysisDataService::Instance().retrieve("out"));
+
+    TS_ASSERT(mws);
+    TS_ASSERT_EQUALS(mws->getNumberHistograms(), 1);
+    TS_ASSERT(!mws->isHistogramData());
+    TS_ASSERT_EQUALS(mws->blocksize(), tws->rowCount());
+
+    auto &X = mws->x(0);
+    auto &Y = mws->y(0);
+    auto &E = mws->e(0);
+
+    for (size_t i = 0; i < tws->rowCount(); ++i) {
+      TableRow row = tws->getRow(i);
+      std::string x;
+      double y, e;
+      row >> x >> y >> e;
+      TS_ASSERT_EQUALS(boost::lexical_cast<double>(x), X[i]);
```

Comment:
You are trying to convert an `std::string` to a `double`. Your method is correct and works, but since `C++11` you can use a whole new family of functions which allow you to perform similar conversions to `lexical_cast`. The functions are `std::stod`, etc. An example would be:
```cpp
  std::string doubleAsString = "123.123";
  auto doubleAsDouble = std::stod(doubleAsString);
```
It is best to use as much as possible form the standard library.